app/services/actions/prospective_student/registration.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Actions
+  module ProspectiveStudent
+    class Registration
+      Error = Class.new(StandardError) do
+        attr_accessor :details
+      end
+
+      attr_reader :prospective
+
+      def initialize(prospective)
+        @prospective = prospective
+      end
+
+      class << self
+        def call(prospective)
+          new(prospective).call
+        end
+      end
+
+      def call
+        valid!
+
+        ActiveRecord::Base.transaction do
+          user.save!
+          student.save!
+          register!
+          archive!
+          upload_avatar!
+        end
+
+        true
+      end
+
+      private
+
+      delegate :unit,
+               :can_permanently_register?,
+               :id_number,
+               :email,
+               :obs_status,
+               :academic_term,
+               :student_entrance_type,
+               :avatar,
+               to: :prospective
+
+      def valid!
+        return true if user.valid? && student.valid?
+
+        error         = Error.new 'RecordInvalid'
+        error.details = [user.errors.full_messages, student.errors.full_messages].flatten!
+
+        raise error
+      end
+
+      def register!
+        prospective.update!(registered: true)
+      end
+
+      def archive!
+        prospective.update!(archived: true) if user.activated
+      end
+
+      def upload_avatar!
+        return if avatar.blank?
+
+        file = Tempfile.new(['avatar', '.jpg'])
+        file.write(Base64.decode64(avatar).force_encoding('UTF-8'))
+        user.avatar.attach(io: file.open, filename: "#{id_number}.jpg")
+        file.unlink
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def student
```

Comment:
```suggestion
      def student # rubocop:disable Metrics/MethodLength
```

Bu blokta disable ediyor (blok sonunda enable'a gerek kalmıyor).  Başka yerler de olabilir, hemen gözüme iliştiğinden yazdım.